swhks/src/main.rs

```diff
@@ -60,18 +61,50 @@ fn main() -> std::io::Result<()> {
     let _ = nix::unistd::daemon(true, false);
 
     setup_swhks(invoking_uid, PathBuf::from(runtime_dir));
-    loop {
-        if let Ok(mut stream) = UnixStream::connect(&sock_file_path) {
-            // Only if the environment has changed, send it to swhkd, else do nothing.
-            if prev_hash != calculate_hash(get_env().unwrap()) {
-                log::debug!("Env changed, sending to swhkd");
-                let new_env = get_env().unwrap();
-                let _ = stream.write_all(new_env.as_bytes());
-                log::debug!("Sent env to swhkd");
-                prev_hash = calculate_hash(new_env);
+
+    if Path::new(&sock_file_path).exists() {
+        fs::remove_file(&sock_file_path)?;
+    }
+
+    let listener = UnixListener::bind(sock_file_path)?;
+    let mut buff = [0; 1];
+    log::debug!("Listening for incoming connections...");
+
+    for stream in listener.incoming() {
+        match stream {
+            Ok(mut stream) => {
+                println!("Connection established!");
+                stream.read_exact(&mut buff)?;
+                println!("Received: {:?}", buff);
+                if buff == [1] {
+                    log::debug!("Received VERIFY message from swhkd");
+                    let _ = stream.write_all(prev_hash.to_string().as_bytes());
+                    log::debug!("Sent hash to swhkd");
+                    stream.flush()?;
+                    continue;
+                }
+                if buff == [2] {
```

Comment:
Need better documentation in these branches. Possibly best to just wrap this up in a utils / communication file and keep it separate from the main source file. Causes unecessary confusion for incoming contributors. Lets discuss further over call when you're free.